ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -149,6 +158,11 @@ private OutputNode buildOutputNode(final PlanNode sourcePlanNode) {
 
     final Into intoDataSource = analysis.getInto().get();
 
+    if (ksqlConfig.getBoolean(KsqlConfig.KSQL_ANY_KEY_NAME_ENABLED)) {
+      ((VerifiableNode) sourcePlanNode)
+          .validateKeyPresent(intoDataSource.getName());
+    }
```

Comment:
This is the call to throw, should the projection not include the key columns.

Once the `any key name` functionality has been turned on, this check can move into `KsqlStructuredDataOutputNode` to make it more OO.